src/volume_mgr.rs

```diff
@@ -319,13 +353,55 @@ where
                 return Err(Error::VolumeStillInUse);
             }
         }
-
         let volume_idx = self.get_volume_by_id(volume)?;
+        if !self.open_volumes[volume_idx].read_only {
+            let VolumeType::Fat(volume_type) = &self.open_volumes[volume_idx].volume_type;
```

Comment:
I've tried to write this code on the assumption that a non-FAT filesystem will be added in the future. The irrefutable match here will break when that happens - I think `set_volume_status_dirty` doesn't need `volume_type` as an argument - it can just work it out.

Reply:
I changed this